tower-http/src/sensitive_header.rs

```diff
@@ -0,0 +1,224 @@
+    /// Create a new [`SetSensitiveRequestHeaderLayer`].
+    pub fn new(header: HeaderName) -> Self {
+        Self { header }
+    }
+}
+
+impl<S> Layer<S> for SetSensitiveRequestHeaderLayer {
+    type Service = SetSensitiveRequestHeader<S>;
+
+    fn layer(&self, inner: S) -> Self::Service {
+        SetSensitiveRequestHeader {
+            inner,
+            header: self.header.clone(),
+        }
+    }
+}
+
+/// Mark a request header as [sensitive].
+///
+/// [sensitive]: https://docs.rs/http/latest/http/header/struct.HeaderValue.html#method.set_sensitive
+#[derive(Clone, Debug)]
+pub struct SetSensitiveRequestHeader<S> {
+    inner: S,
+    header: HeaderName,
+}
+
+impl<S> SetSensitiveRequestHeader<S> {
+    /// Create a new [`SetSensitiveRequestHeader`].
+    pub fn new(inner: S, header: HeaderName) -> Self {
+        Self { inner, header }
+    }
+
+    /// Gets a reference to the underlying service.
+    pub fn get_ref(&self) -> &S {
+        &self.inner
+    }
+
+    /// Gets a mutable reference to the underlying service.
+    pub fn get_mut(&mut self) -> &mut S {
+        &mut self.inner
+    }
+
+    /// Consumes `self`, returning the underlying service.
+    pub fn into_inner(self) -> S {
+        self.inner
+    }
+}
+
+impl<ReqBody, ResBody, S> Service<Request<ReqBody>> for SetSensitiveRequestHeader<S>
+where
+    S: Service<Request<ReqBody>, Response = Response<ResBody>>,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = S::Future;
+
+    #[inline]
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.inner.poll_ready(cx)
+    }
+
+    fn call(&mut self, mut req: Request<ReqBody>) -> Self::Future {
+        if let Some(value) = req.headers_mut().get_mut(&self.header) {
+            value.set_sensitive(true);
+        }
+
+        self.inner.call(req)
+    }
+}
+
+/// Mark a response header as [sensitive].
+///
+/// [sensitive]: https://docs.rs/http/latest/http/header/struct.HeaderValue.html#method.set_sensitive
+#[derive(Clone, Debug)]
+pub struct SetSensitiveResponseHeaderLayer {
+    header: HeaderName,
+}
+
+impl SetSensitiveResponseHeaderLayer {
+    /// Create a new [`SetSensitiveResponseHeaderLayer`].
+    pub fn new(header: HeaderName) -> Self {
+        Self { header }
+    }
+}
+
+impl<S> Layer<S> for SetSensitiveResponseHeaderLayer {
+    type Service = SetSensitiveResponseHeader<S>;
+
+    fn layer(&self, inner: S) -> Self::Service {
+        SetSensitiveResponseHeader {
+            inner,
+            header: self.header.clone(),
+        }
+    }
+}
+
+/// Mark a response header as [sensitive].
+///
+/// [sensitive]: https://docs.rs/http/latest/http/header/struct.HeaderValue.html#method.set_sensitive
+#[derive(Clone, Debug)]
+pub struct SetSensitiveResponseHeader<S> {
+    inner: S,
+    header: HeaderName,
+}
+
+impl<S> SetSensitiveResponseHeader<S> {
+    /// Create a new [`SetSensitiveResponseHeader`].
```

Comment:
nit, take it or leave it:
```suggestion
    /// Create a new [`SetSensitiveResponseHeader`] service.
```